pkg/graveler/graveler.go

```diff
@@ -1872,7 +1873,7 @@ func (g *Graveler) addCommitNoLock(ctx context.Context, repository *RepositoryRe
 }
 
 func (g *Graveler) isStagingEmpty(ctx context.Context, repository *RepositoryRecord, branch *Branch) (bool, error) {
-	itr, err := g.listStagingArea(ctx, branch)
+	itr, err := g.listStagingArea(ctx, branch, 0)
```

Comment:
```suggestion
	itr, err := g.listStagingArea(ctx, branch, 1)
```

Reply:
Thanks - good call to optimize empty checks too